pkg/apis/core/validation/validation.go

```diff
@@ -5195,17 +5196,31 @@ func ValidatePodTemplateSpec(spec *core.PodTemplateSpec, fldPath *field.Path, op
 	return allErrs
 }
 
-func ValidateReadOnlyPersistentDisks(volumes []core.Volume, fldPath *field.Path) field.ErrorList {
+// ValidateReadOnlyPersistentDisks validates pd reads and writes
+func ValidateReadOnlyPersistentDisks(featureGate featuregate.FeatureGate, volumes, oldVolumes []core.Volume, fldPath *field.Path, isNew bool) field.ErrorList {
 	allErrs := field.ErrorList{}
+
+	if !featureGate.Enabled(features.SkipReadOnlyValidationGCE) {
+		return field.ErrorList{}
+	}
+
+	isWriteablePD := func(vol *core.Volume) bool {
+		return vol.GCEPersistentDisk != nil && !vol.GCEPersistentDisk.ReadOnly
+	}
+
+	if !isNew {
```

Comment:
Do you need `isNew` ?  A `range` loop on a nil slice will run 0 times.

Reply:
If it is not a new use for reading and writing, allErrs is returned

Reply:
`isNew` is the same as `len(oldVolumes) > 0`, so `range` should cover it without the need for `isNew`.